src/server.rs

```diff
@@ -31,9 +31,11 @@ pub async fn run(cli: Cli) -> anyhow::Result<()> {
     let provider = Provider::<Http>::try_from(evm_rpc_url)?;
     let chain_id = provider.get_chainid().await?.as_u64();
     let wallet = LocalWallet::from_bytes(&private_key)?.with_chain_id(chain_id);
-    let provider_with_nonce = NonceManagerMiddleware::new(provider, wallet.address());
-    let client: DefaultSignerMiddleware = SignerMiddleware::new(provider_with_nonce, wallet);
+
+    let nonce_manager = Arc::new(NonceManagerMiddleware::new(provider, wallet.address()));
```

Comment:
This extra layer of `Arc` seems potentially unnecessary since the `NonceManager` gets wrapped in `SignerMiddleware`, which is then wrapped in an `Arc`.